### PR TITLE
Water-2803 - Get season for abstraction period

### PR DESCRIPTION
**Pull request by Stephan-Hough**
feat/2803
-- new helper to get the abstraction period season

### CODE REVIEW

File: src/returns/date/season-in-abstraction-period.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+'use-strict';
+
+const moment = require('moment');
+const summer = {
+  startDay: '01',
```
Comment (james-siteclick) [inline src/returns/date/season-in-abstraction-period.js:5]: Do these need to be strings or can they be integers?

File: src/returns/date/season-in-abstraction-period.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+const getAbstractionPeriodSeason = (abstractionPeriod) => {
+  // For the season to be summer, this abstraction period must
+  // sit within the summer period (01/04 - 31/10)
+  if (isWithinAbstractionPeriod(abstractionPeriod, summer)) {
+    return CHARGE_SEASON.summer;
+  }
+
+  // For the season to be winter, this abstraction period must
+  // sit within the winter period (01/11 - 31/03)
+  if (this.isWithinAbstractionPeriod(abstractionPeriod, winter)) {
```
Comment (james-siteclick) [inline src/returns/date/season-in-abstraction-period.js:73]: Should this be:
```
if (isWithinAbstractionPeriod(abstractionPeriod, winter)) { ...
```
i.e. without the `this.`

File: src/returns/date/season-in-abstraction-period.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    return CHARGE_SEASON.winter;
+  }
+
+  return CHARGE_SEASON.allYear;
```
Comment (james-siteclick) [inline src/returns/date/season-in-abstraction-period.js:77]: It would be good to have a test case for "all year"

**Review by james-siteclick — CHANGES_REQUESTED** [review]

File: src/returns/date/season-in-abstraction-period.js (hunk 1 above)
Comment (Stephan-Hough) [reply src/returns/date/season-in-abstraction-period.js:5]: done

File: src/returns/date/season-in-abstraction-period.js (hunk 2 above)
Comment (Stephan-Hough) [reply src/returns/date/season-in-abstraction-period.js:73]: done

File: src/returns/date/season-in-abstraction-period.js (hunk 3 above)
Comment (Stephan-Hough) [reply src/returns/date/season-in-abstraction-period.js:77]: done

**Review by james-siteclick — APPROVED** [review]

**Pull request merged**